commit 48a4c10a413cef66409421fbc11811fb78f7f894
Author: Bruno Randolf <br1@einfach.org>
Date:   Mon Jan 8 12:29:01 2018 +0000

    Add retries
    
    from https://sourceforge.net/p/dfu-util/tickets/46/

diff --git a/src/dfuse.c b/src/dfuse.c
index 7e4c7e2..95b133f 100644
--- a/src/dfuse.c
+++ b/src/dfuse.c
@@ -179,6 +179,7 @@ int dfuse_special_command(struct dfu_if *dif, unsigned int address,
 	struct dfu_status dst;
 	int firstpoll = 1;
 	int zerotimeouts = 0;
+	int retries = 0;
 
 	if (command == ERASE_PAGE) {
 		struct memsegment *segment;
@@ -223,8 +224,14 @@ int dfuse_special_command(struct dfu_if *dif, unsigned int address,
 			dfuse_command_name[command]);
 	}
 	do {
+retry:
 		ret = dfu_get_status(dif, &dst);
 		if (ret < 0) {
+			if (++retries < 10) {
+				usleep(10);
+				goto retry;
+			}
+
 			errx(EX_IOERR, "Error during special command \"%s\" get_status",
 			     dfuse_command_name[command]);
 		}
@@ -264,6 +271,7 @@ int dfuse_dnload_chunk(struct dfu_if *dif, unsigned char *data, int size,
 	int bytes_sent;
 	struct dfu_status dst;
 	int ret;
+	int retries = 0;
 
 	ret = dfuse_download(dif, size, size ? data : NULL, transaction);
 	if (ret < 0) {
@@ -273,8 +281,14 @@ int dfuse_dnload_chunk(struct dfu_if *dif, unsigned char *data, int size,
 	bytes_sent = ret;
 
 	do {
+retry:
 		ret = dfu_get_status(dif, &dst);
 		if (ret < 0) {
+			if (++retries < 10) {
+				usleep(10);
+				goto retry;
+			}
+
 			errx(EX_IOERR, "Error during download get_status");
 			return ret;
 		}
