diff --git a/keyboards/handwired/tzarc/cyclone/config.h b/keyboards/handwired/tzarc/cyclone/config.h
index f30a3be16..7f1894fed 100644
--- a/keyboards/handwired/tzarc/cyclone/config.h
+++ b/keyboards/handwired/tzarc/cyclone/config.h
@@ -67,7 +67,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #define PIN_COL_LATCH_3V3 A4
 #define PIN_COL_CLK_3V3 B3
-#define PIN_COL_DATA_3V3 A15
+#define PIN_COL_DATA_3V3 B1
 
 #define PIN_WS2812_3V3 A2
 
diff --git a/keyboards/handwired/tzarc/cyclone/cyclone.c b/keyboards/handwired/tzarc/cyclone/cyclone.c
index 0957bf6cb..0d09a66ac 100644
--- a/keyboards/handwired/tzarc/cyclone/cyclone.c
+++ b/keyboards/handwired/tzarc/cyclone/cyclone.c
@@ -45,7 +45,33 @@ led_config_t g_led_config = {{// Key Matrix to LED Index
                               4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4}};
 #endif  // RGB_MATRIX_ENABLE
 
+/*
+ * Half-speed SPI configuration (8MHz, CPHA=0, CPOL=0, MSb first).
+ */
+static const SPIConfig hs_spicfg = {NULL, GPIOA, 1, SPI_CR1_BR_0, 0};
+
+static uint32_t last = 0;
+void            matrix_scan_user(void) {
+    uint32_t now = timer_read32();
+    if (now - last > 20) {
+        const char txbuf[4] = {0x12, 0x34, 0x56, 0x78};
+        char       rxbuf[4] = {0};
+        last                = now;
+
+        spiStart(&SPID1, &hs_spicfg);         /* Setup transfer parameters.       */
+        spiSelect(&SPID1);                    /* Slave Select assertion.          */
+        spiExchange(&SPID1, 4, txbuf, rxbuf); /* Atomic transfer operations.      */
+        spiUnselect(&SPID1);                  /* Slave Select de-assertion.       */
+    }
+}
+
 void matrix_init_kb(void) {
+    palSetPadMode(GPIOA, 5, PAL_MODE_ALTERNATE(0) | PAL_STM32_OSPEED_HIGHEST); /* New SCK.     */
+    palSetPadMode(GPIOA, 6, PAL_MODE_ALTERNATE(0) | PAL_STM32_OSPEED_HIGHEST); /* New MISO.    */
+    palSetPadMode(GPIOA, 7, PAL_MODE_ALTERNATE(0) | PAL_STM32_OSPEED_HIGHEST); /* New MOSI.    */
+    palSetPad(GPIOA, 1);
+    palSetPadMode(GPIOA, 1, PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_HIGHEST); /* New CS.      */
+
     // Set the level shifter GPIO directions and initial states, needs to be before shift registers get initialised
     setPinOutput(PIN_EN_5V_IO_INV);
     setPinOutput(PIN_EN_5V_IO);
@@ -65,9 +91,9 @@ void keyboard_post_init_kb(void) {
     }
 #endif  // EEPROM_ENABLE
 
-    debug_enable   = false;
-    debug_matrix   = false;
-    debug_keyboard = false;
+    debug_enable   = true;
+    debug_matrix   = true;
+    debug_keyboard = true;
     debug_mouse    = false;
 
 #ifdef TEST_AUDIO_PIN_OUTPUT
diff --git a/keyboards/handwired/tzarc/cyclone/rules.mk b/keyboards/handwired/tzarc/cyclone/rules.mk
index 4378430a7..4d5e0eee1 100644
--- a/keyboards/handwired/tzarc/cyclone/rules.mk
+++ b/keyboards/handwired/tzarc/cyclone/rules.mk
@@ -36,9 +36,9 @@ RAW_ENABLE = yes
 RGB_MATRIX_ENABLE = WS2812
 WS2812_DRIVER = bitbang
 
-EEPROM_DRIVER = i2c
+#EEPROM_DRIVER = i2c
 
-OLED_DRIVER_ENABLE = yes
+#OLED_DRIVER_ENABLE = yes
 # AUDIO_ENABLE = yes
 # LINK_TIME_OPTIMIZATION_ENABLE = yes
 
